crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -165,7 +165,7 @@ where
             let Some(body_index) = provider.block_body_indices(block_number)? else {
                 return Ok(None)
             };
-            let tx_index = id - body_index.last_tx_num();
+            let tx_index = body_index.last_tx_num() - id;
             Ok(Some((None, tx_index as usize)))
```

Comment:
> unsure on what good is a `Ok(Some(None,txindex))`

no good, let's do
```suggestion
            Ok(Some(None))
```
and return `Some((block_state, tx_index))` below